src/components/ReferralHeroSection/ReferralHeroSection.spec.tsx

```diff
@@ -0,0 +1,150 @@
+/* eslint-disable @typescript-eslint/unbound-method */
+import React from 'react'
+import { fireEvent, waitFor, act } from '@testing-library/react'
+import { t } from 'decentraland-dapps/dist/modules/translation/utils'
+import { config } from '../../modules/config'
+import { locations } from '../../modules/routing/locations'
+import { renderWithProviders } from '../../tests/tests'
+import {
+  REFERRAL_CONTAINER_TEST_ID,
+  REFERRAL_COPY_OPTION_TEST_ID,
+  REFERRAL_INPUT_TEST_ID,
+  REFERRAL_SHARE_BUTTON_TEST_ID,
+  REFERRAL_SHARE_MENU_TEST_ID,
+  REFERRAL_SHARE_X_OPTION_TEST_ID,
+  REFERRAL_STEPS_CONTAINER_TEST_ID
+} from './constants'
+import { ReferralHeroSection } from './ReferralHeroSection'
+import { Props } from './ReferralHeroSection.types'
+
+const INVITE_REFERRER_URL = config.get('INVITE_REFERRER_URL', '')
+
+jest.mock('../../modules/routing/locations', () => ({
+  locations: {
+    twitter: jest.fn()
+  }
+}))
+
+describe('ReferralHeroSection', () => {
+  const profileAddress = '0x123456789abcdef'
+  const inviteUrl = `${INVITE_REFERRER_URL}/${profileAddress}`
+
+  const renderReferralHeroSection = (props: Partial<Props> = {}) => {
+    return renderWithProviders(<ReferralHeroSection profileAddress={profileAddress} isLoading={false} {...props} />)
+  }
+
+  beforeEach(() => {
+    jest.useFakeTimers()
+    Object.assign(navigator, {
+      clipboard: {
+        writeText: jest.fn().mockImplementation(() => Promise.resolve())
+      }
+    })
+  })
+
+  afterEach(() => {
+    jest.useRealTimers()
+    jest.clearAllMocks()
+  })
+
+  describe('when the component renders', () => {
+    it('should display title and subtitle', () => {
+      const { getByText } = renderReferralHeroSection()
+      expect(getByText(t('referral_hero_section.title'))).toBeInTheDocument()
+      expect(getByText(t('referral_hero_section.subtitle'))).toBeInTheDocument()
+    })
+
+    it('should display the three steps', () => {
+      const { getByTestId, getByText } = renderReferralHeroSection()
+      expect(getByTestId(REFERRAL_STEPS_CONTAINER_TEST_ID)).toBeInTheDocument()
+      expect(getByText(t('referral_hero_section.step_1'))).toBeInTheDocument()
+      expect(getByText(t('referral_hero_section.step_2'))).toBeInTheDocument()
+      expect(getByText(t('referral_hero_section.step_3'))).toBeInTheDocument()
+    })
+
+    it('should display the referral container when not loading', () => {
```

Comment:
What do you think about splitting these tests into two sections? The first one is related to the render checks you want to do and are related to the default rendering (under the `when the component renders`), and another two sections that describe two other contexts, when the component is loading and when the component isn't.
```suggestion
    describe('when the component is not loading', () => {
      let renderedComponent: ReturnType<typeof renderReferralHeroSection>
      beforeEach(() => {
        renderedComponent = renderReferralHeroSection({ isLoading: false })
      })
      it('should display the referral container', () => {
```
The same goes for the other test.